Sources/TuistCore/Carthage/CarthageGraphLoader.swift

```diff
@@ -0,0 +1,61 @@
+import Foundation
+import TSCBasic
+import TuistSupport
+
+public enum CarthageGraphLoaderError: FatalError, Equatable {
```

Comment:
What do you think about the idea of having a generic graph loader for dependencies like we are doing for [the project graph](https://github.com/tuist/tuist/blob/149a442dbfa58d6ad1218a18b817b59be61cae8a/Sources/TuistCore/Graph/GraphLoader.swift#L5).

```swift
protocol DependenciesGraphLoading {
  func load(at path: AbsolutePath) throws -> Graph
}
```

Then we can extend this loader to account for dependencies defined by other dependency managers like SPM.

Reply:
I like the idea. 
I thought about starting in that direction but also thought that it would be wiser to go from specific to general.

I can add a reference to this PR in the code as a reminder. When we tackle the next dependency manager I think we will have a better picture to create some abstractions.

What do you think?